src/lib/autocomplete/autocomplete.ts

```diff
@@ -77,7 +77,18 @@ export class MatAutocomplete implements AfterContentInit {
   @ContentChildren(MatOptgroup) optionGroups: QueryList<MatOptgroup>;
 
   /** Function that maps an option's control value to its display value in the trigger. */
-  @Input() displayWith: ((value: any) => string) | null = null;
+  @Input()
+  get displayWith() {
+    return this._displayWith;
+  }
+  set displayWith(value: any) {
+    this._displayWith = typeof value === 'function' ? value : null;
+    this.displayWithChange.emit();
+  }
+  private _displayWith: ((value: any) => string) | null = null;
+
+  /** @docs-private */
+  readonly displayWithChange = new EventEmitter<void>();
```

Comment:
Can yo rename this to `_displayWithChange`, add a doc string and switch it to a Subject? Also you'll be able to remove the `@docs-private` when it's prefixed with an underscore.

Reply:
Done.